testsupport/metrics.go

```diff
@@ -0,0 +1,60 @@
+package testsupport
+
+import (
+	"time"
+)
+
+// MetricsAssertionHelper stores baseline metric values when initialized and has convenient functions for metrics assertions
+type MetricsAssertionHelper struct {
+	await          metricsProvider
+	baselineValues map[string]float64
+}
+
+type metricKey string
```

Comment:
since you don't declare the particular metric constants as the `metricKey` type, then I don't see any reason why we should declare the type at all - you can remove it...

Reply:
Oops, yes I was originally thinking to use it but then figured it's not necessary. Forgot to remove it. Done